qiskit/primitives/base/base_estimator.py

```diff
@@ -254,3 +325,43 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2:
+    """Estimator base class version 2.
+
+    An estimator estimates expectation values for provided quantum circuit and
+    observable combinations.
+    """
+
+    def __init__(self, precision: float | None):
```

Comment:
I'd prefer to force this into a numeric value unless there is a reason to have a None state: it would be nice to avoid `is None` checks where possible.

```suggestion
    def __init__(self, precision: float):
```

(likewise in `@property`)